test/e2e_new/common.go

```diff
@@ -0,0 +1,70 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_new
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	corev1 "k8s.io/api/core/v1"
+	"path/filepath"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/util"
+)
+
+func setupSpecNamespace(ctx context.Context, specName string, clusterProxy framework.ClusterProxy, artifactFolder string) (*corev1.Namespace, context.CancelFunc) {
```

Comment:
Wondering if we should add those methods to the framework, it seems those are the same of what we have in CAPI...

Reply:
I certainly agree. setupNamespace can become a helper method. dumpSpecResourcesAndCleanup does a couple of things so unsure if that should be a single helper. I understand the helper methods in the framework as helpers to interact with the cluster (getting, setting and waiting for state)

Reply:
Will revisit what should be exported as common functions once we have more providers (CAPV) using the framework.